sonar-db/src/main/java/org/sonar/db/charset/MysqlCharsetHandler.java

```diff
@@ -45,37 +48,42 @@ protected MysqlCharsetHandler(SqlExecutor selectExecutor) {
   }
 
   @Override
-  void handle(Connection connection, boolean enforceUtf8) throws SQLException {
-    logInit(enforceUtf8);
-    checkCollation(connection, enforceUtf8);
+  void handle(Connection connection, Set<DatabaseCharsetChecker.Flag> flags) throws SQLException {
+    logInit(flags);
+    checkCollation(connection, flags);
   }
 
-  private static void logInit(boolean enforceUtf8) {
+  private static void logInit(Set<DatabaseCharsetChecker.Flag> flags) {
     String message = "Verify that database collation is case-sensitive";
-    if (enforceUtf8) {
+    if (flags.contains(ENFORCE_UTF8)) {
       message = "Verify that database collation is UTF8";
     }
     LOGGER.info(message);
   }
 
-  private void checkCollation(Connection connection, boolean enforceUtf8) throws SQLException {
+  private void checkCollation(Connection connection, Set<DatabaseCharsetChecker.Flag> flags) throws SQLException {
     // All VARCHAR columns are returned. No need to check database general collation.
     // Example of row:
     // issues | kee | utf8 | utf8_bin
     List<ColumnDef> columns = select(connection,
       ColumnDef.SELECT_COLUMNS +
         "FROM INFORMATION_SCHEMA.columns " +
         "WHERE table_schema=database() and character_set_name is not null and collation_name is not null", ColumnDef.ColumnDefRowConverter.INSTANCE);
-    List<String> utf8Errors = new ArrayList<>();
+    Set<String> errors = new HashSet<>();
```

Comment:
using a `List` for errors in `MssqlCharsetHandler`, using a `Set` here, you should choose which side you're on :)